src/components/ReferralProgramCTA.tsx

```diff
@@ -1,32 +1,45 @@
 import React from 'react';
+import {withOnyx} from 'react-native-onyx';
 import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
+import * as User from '@userActions/User';
 import CONST from '@src/CONST';
 import Navigation from '@src/libs/Navigation/Navigation';
+import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {DismissedReferralBanners} from '@src/types/onyx/Account';
 import Icon from './Icon';
 import {Close} from './Icon/Expensicons';
 import {PressableWithoutFeedback} from './Pressable';
 import Text from './Text';
 import Tooltip from './Tooltip';
 
-type ReferralProgramCTAProps = {
+type ReferralProgramCTAOnyxProps = {
+    dismissedReferralBanners: DismissedReferralBanners;
+};
+
+type ReferralProgramCTAProps = ReferralProgramCTAOnyxProps & {
     referralContentType:
         | typeof CONST.REFERRAL_PROGRAM.CONTENT_TYPES.MONEY_REQUEST
         | typeof CONST.REFERRAL_PROGRAM.CONTENT_TYPES.START_CHAT
         | typeof CONST.REFERRAL_PROGRAM.CONTENT_TYPES.SEND_MONEY
         | typeof CONST.REFERRAL_PROGRAM.CONTENT_TYPES.REFER_FRIEND;
-
-    /** Method to trigger when pressing close button of the banner */
-    onCloseButtonPress?: () => void;
 };
 
-function ReferralProgramCTA({referralContentType, onCloseButtonPress = () => {}}: ReferralProgramCTAProps) {
+function ReferralProgramCTA({referralContentType, dismissedReferralBanners}: ReferralProgramCTAProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const theme = useTheme();
 
+    const handleDismissCallToAction = () => {
+        User.dismissReferralBanner(CONST.REFERRAL_PROGRAM.CONTENT_TYPES.REFER_FRIEND);
```

Comment:
This should have been `referralContentType` instead of `CONST.REFERRAL_PROGRAM.CONTENT_TYPES.REFER_FRIEND` to dismiss the correct banner